analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/ProjectBuilder.cs

```diff
@@ -72,9 +72,25 @@ public ProjectBuilder AddDocument(string path)
         {
             _ = path ?? throw new ArgumentNullException(nameof(path));
             var fileInfo = new FileInfo(path);
+            var relativePathFromTestCases = RelativePathFromTestCases(fileInfo);
             return fileInfo.Extension == fileExtension
-                ? AddDocument(project, fileInfo.Name, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
+                ? AddDocument(project, relativePathFromTestCases, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
                 : throw new ArgumentException($"The file extension '{fileInfo.Extension}' does not match the project language '{project.Language}'.", nameof(path));
+
+            static string RelativePathFromTestCases(FileInfo fileInfo)
+            {
+                Stack<string> directories = new();
+                directories.Push(fileInfo.Name);
+                var directory = fileInfo.Directory;
+                while (directory != null && directory.Name.ToUpper() != "TESTCASES")
+                {
+                    directories.Push(directory.Name);
+                    directory = directory.Parent;
+                }
+                return directory == null
+                    ? throw new ArgumentException("path must contain 'TestCases'", nameof(fileInfo))
+                    : Path.Combine(directories.ToArray());
```

Comment:
I went this way to prevent any OS-specific path separator issues. `LastIndexOf` with `Path.PathSeparator` should also work, though.

Reply:
Being OS indenependent is not a requirement. There are many other places where UTs would fail on unix OS.

Let's just use this, it's way more simple and streight-forward
```
        public ProjectBuilder AddDocument(string path)
        {
            const string TestCases = @"TestCases\";
            _ = path ?? throw new ArgumentNullException(nameof(path));
            var fileInfo = new FileInfo(path);
            var index = fileInfo.FullName.IndexOf(TestCases);
            var relativePathFromTestCases = index < 0
                ? throw new ArgumentException($"{nameof(path)} must contain '{TestCases}'", nameof(path))
                : fileInfo.FullName.Substring(index + TestCases.Length);
            return fileInfo.Extension == fileExtension
                ? AddDocument(project, relativePathFromTestCases, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
                : throw new ArgumentException($"The file extension '{fileInfo.Extension}' does not match the project language '{project.Language}'.", nameof(path));
        }
```
If we ever need to build it on unix, we can just use the PathSeparator.